exp/controllers/azuremachinepool_reconciler.go

```diff
@@ -57,6 +57,9 @@ func (s *azureMachinePoolService) Reconcile(ctx context.Context) error {
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "controllers.azureMachinePoolService.Reconcile")
 	defer done()
 
+	// Ensure that the deprecated networking field values have been migrated to the new NetworkInterfaces field.
+	s.scope.AzureMachinePool.SetNetworkInterfacesDefaults()
+
```

Comment:
Defaulter should take care of this, might not need it here.

Reply:
Same as above. It is possible that the reconciler runs before a resource has been modified and therefore before the defaulter has run. This is a protection against that case. The code assumes those new fields are populated and we will cause the reconciler to panic with nil pointer references if a resource has not yet had these fields set.